src/db/add_package.rs

```diff
@@ -221,13 +221,19 @@ fn convert_dependencies(pkg: &MetadataPackage) -> Vec<(String, String, String)>
 fn get_features(pkg: &MetadataPackage) -> Vec<Feature> {
     let mut features = Vec::with_capacity(pkg.features.len());
     if let Some(subfeatures) = pkg.features.get("default") {
-        features.push(Feature::new("default".into(), subfeatures.clone()));
+        features.push(Feature::new(
+            "default".into(),
+            subfeatures.clone(),
+            Some(false),
```

Comment:
So, this is making me regret using postgres types. `is_optional_dependency` should be non-null when being added, but nullable when being retrieved from the database. Do you know a good way to do that?

Reply:
`fn new` at least could take `bool` and implicitly wrap it into `Some`, so to set it to `None` you'd have to bypass the function and construct a `Feature` instance directly.